core/src/main/java/nl/nn/adapterframework/pipes/AbstractPipe.java

```diff
@@ -83,7 +84,7 @@
  *
  * @see PipeLineSession
  */
-public abstract class AbstractPipe extends TransactionAttributes implements IExtendedPipe, EventThrowing, ApplicationContextAware {
+public abstract class AbstractPipe extends TransactionAttributes implements IExtendedPipe, EventThrowing, ApplicationContextAware, IWithParameters {
```

Comment:
All pipes extend `AbstractPipe` so all pipes implement `IExtendedPipe` and `IPipe` and now also all pipes implement `IWithParameters`.
That makes me wonder if perhaps the interface `IPipe` should extend `IWithParameters` instead of `AbstractPipe implements IWithParameters`?

Reply:
Yea, we need to sort this out, this was purely a quick fix for Proxy classes that can only 'proxy' interface methods.
We also need to sort out what to do with IExcendedPipe and IPipe. I've created #5957